pkg/extractor/index_test.go

```diff
@@ -74,43 +75,39 @@ func TestIndexExtractorListAllBundles(t *testing.T) {
 			"quay.io/osd-addons/gpu-operator-nfd-operator-bundle:4.8.0-0ddc381",
 		},
 	}
-	expectedBundleImages := map[string]bool{
-		"quay.io/osd-addons/gpu-operator-bundle:1.7.1-0ddc381":              true,
-		"quay.io/osd-addons/gpu-operator-bundle:1.8.0-0ddc381":              true,
-		"quay.io/osd-addons/gpu-operator-bundle:1.8.2-0ddc381":              true,
-		"quay.io/osd-addons/gpu-operator-bundle:1.8.3-0ddc381":              true,
-		"quay.io/osd-addons/gpu-operator-bundle:1.9.0-beta-0ddc381":         true,
-		"quay.io/osd-addons/gpu-operator-nfd-operator-bundle:4.8.0-0ddc381": true,
+	expectedBundleImages := []string{
+		"quay.io/osd-addons/gpu-operator-bundle:1.7.1-0ddc381",
+		"quay.io/osd-addons/gpu-operator-bundle:1.8.0-0ddc381",
+		"quay.io/osd-addons/gpu-operator-bundle:1.8.2-0ddc381",
+		"quay.io/osd-addons/gpu-operator-bundle:1.8.3-0ddc381",
+		"quay.io/osd-addons/gpu-operator-bundle:1.9.0-beta-0ddc381",
+		"quay.io/osd-addons/gpu-operator-nfd-operator-bundle:4.8.0-0ddc381",
 	}
 	cache := NewIndexMemoryCache()
 	extractor := NewIndexExtractor(WithIndexCache(cache))
 
 	// test bundleImages for all packages are listed
 	bundleImages, err := extractor.ExtractAllBundleImages(indexImage)
 	require.NoError(t, err)
-	require.True(t, allBundleImagesListed(bundleImages, expectedBundleImages))
+	requireContainsAll(t, bundleImages, expectedBundleImages)
+
 	cachedBundleImages := cache.GetBundleImages(indexImage, allBundlesKey)
-	require.True(t, allBundleImagesListed(cachedBundleImages, expectedBundleImages))
+	requireContainsAll(t, cachedBundleImages, expectedBundleImages)
 
 	// test bundles have been cached per pkgName
 	for pkgName, expectedBundleImages := range bundleImagesMap {
 		bundleImages, err := extractor.ExtractBundleImages(indexImage, pkgName)
 		require.NoError(t, err)
-		require.Equal(t, bundleImages, expectedBundleImages)
+		requireContainsAll(t, bundleImages, expectedBundleImages)
+
 		cachedBundleImages := cache.GetBundleImages(indexImage, pkgName)
-		require.Equal(t, cachedBundleImages, expectedBundleImages)
+		requireContainsAll(t, cachedBundleImages, expectedBundleImages)
 	}
 }
 
-// cache or indexImage list bundles in different order, using a map to avoid flakiness
-func allBundleImagesListed(bundleImages []string, expectedBundleImages map[string]bool) bool {
-	if len(bundleImages) != len(expectedBundleImages) {
-		return false
-	}
-	for _, image := range bundleImages {
-		if _, ok := expectedBundleImages[image]; !ok {
-			return false
-		}
+func requireContainsAll(t *testing.T, images []string, expectedImages []string) {
```

Comment:
I think`testify` already has this https://pkg.go.dev/github.com/stretchr/testify/require#Assertions.ElementsMatch
Except `ElementsMatch` would fail for `images=[]string{"1","1","1"} expectedImages=[]string{"1","2","3"}` whereas this would not.

Reply:
Was looking for that. Indeed my testing helper had a bug :+1: